### ISSUE TITLE
Feature Request: rmate remote editing support

### ISSUE THREAD

**Issue by jasonnet** [notepad-plus-plus/notepad-plus-plus#8276]
### Description of the feature request
Several text editors ([example](https://medium.com/@prtdomingo/editing-files-in-your-linux-virtual-machine-made-a-lot-easier-with-remote-vscode-6bb98d0639a4)) out there have support for rmate-like ([link](https://github.com/aurora/rmate), [link](https://github.com/textmate/rmate), etc) features that, similar to NppFTP, allow a user to edit remote files, and, unlike NppFTP, allow one to bring the file up in an editor from a SSH 
session on that machine.  This can be HIGHLY  helpful in the workflow of the user.

Npp and NppFTP seem to lack this feature or the ability to create an external hack to support it.
The suggestion for a first step is to provide a way to invoke Npp on the command line with parameters that cause Npp to present the user with an editing session with the specified remote
file within.  If that's in place, the semi-technical community can begin hacking up possible more complete solutions.

I see someone made a similar request [here](https://github.com/ashkulz/NppFTP/issues/244).   The claim there is that the NppFTP can not yet implement this because there is not yet support in Npp to call a plugin from the command line.

### Expected Behavior

For a simple first step:

    C:\Program Files\Notepad++>notepad++.exe -multiInst -openSession -pluginCommand NppFTP joe@myserver.foo.com:.bash.rc
    or 
    C:\Program Files\Notepad++>notepad++.exe -multiInst -openSession -pluginCommand NppFTP myNppFTPProfile:bash.rc

would cause the editor to present an edit session for the intended file after attempting to establish an optional session to that machine and fetch/check-for the current rendition of that file on the remote machine.

### Actual Behavior

Not currently supported.

**Comment by asvc** [comment]
Are you suggesting that text editor should listen on an open port/socket for incoming connections?..

**Comment by jasonnet** [comment]
No.  For now, let's keep this simple.   I'm just suggesting the command line support I suggested above.  No sockets.  

How that's used is up to each external developer that chooses to use it.

**Comment by asvc** [comment]
Could I offer a safer option? Mount whatever endpoint locally and edit the file as usual.

**Comment by jasonnet** [comment]
> Could I offer a safer option? Mount whatever endpoint locally and edit the file as usual.

Hey Andrew.   Thanks for the suggestion.   

Reminder: The request is not for remote editing, despite the issue headline.   That's simply a sexy use case.  The request is for a way to pass commands+parameters to NppFTP from the command line.

As for remote-triggered editing.  I can think of many ways for the requested feature to be leveraged to achieve that goal.  Some of them are just as safe/unsafe as mounting.

Mounting definitely can work for some people, but some of us are not fortunate enough to have only a few predictable remote machines+directories that we access.  If that were the case, then the existing NppFTP would be sufficient and we wouldn't need to resort to Remote VSCode, TM2, Atom,  instead of Npp+NppFTP.  Similarly, some of clients+server pairs are not suitable for file system mounting.

Anyway, thanks for taking time to validate this feature request.
